### PR TITLE
Added a mechanism for explicitly specifying the forwarder target

### PR DESCRIPTION
**Pull request by vitroth**
necessary for standalone configs which will send to a specific indexer not managed by the same chef config

@zquestz

### CODE REVIEW

File: recipes/client.rb [hunk 1]
```diff
@@ -24,7 +24,14 @@
 include_recipe 'chef-splunk::user'
 include_recipe 'chef-splunk::install_forwarder'
 
-if !node['splunk']['splunk_indexer_search'].nil?
+if !node['splunk']['splunk_indexers'].nil?
```
Comment (zquestz) [inline recipes/client.rb]: Normally not required for you to do .nil? For any value other than boolean false you can do.

```ruby
if node['splunk']['splunk_indexers']
end
```

File: recipes/client.rb [hunk 2]
```diff
@@ -24,7 +24,14 @@
...
+  # If we have an array of indexers to forward to, just use it
+  splunk_servers = []
+  node['splunk']['splunk_indexers'].each do |indexer|
+    splunk_servers.push({ 'ipaddress' => indexer })
+  end
+elsif !node['splunk']['splunk_indexer_search'].nil?
+  # If an explicit chef search was provided for finding the indexer, use that
   splunk_servers = search( # ~FC003
```
Comment (zquestz) [inline recipes/client.rb]: What is ~FC003?

File: recipes/client.rb [hunk 3]
```diff
@@ -32,6 +39,7 @@
     |a, b| a.name <=> b.name
   end
 else
+  # Default to searching for splunk_is_server:true in our current environment
   splunk_servers = search( # ~FC003
```
Comment (zquestz) [inline recipes/client.rb]: Above...

File: recipes/client.rb (hunk 2 above)
Comment (vitroth) [inline recipes/client.rb]: FC003 is the Food Critic rule that alerts on using search without verifying you're in a chef-server environment.

**Comment by philpennock** [comment]
Not seeing how the `splunk_servers` ends up with equivalently keyed entries between the search results vs the manual construction, but I'm willing to chalk that up to my head being somewhat fuzzy.  If you've tested it and it works both ways, then this LGTM.

File: recipes/client.rb (hunk 1 above)
Comment (zquestz) [inline recipes/client.rb]: This should have been fixed before merge... this is not nearly as clear with the current syntax.

Comment (vitroth) [inline recipes/client.rb]: Oops, missed that change.  I'll update when I'm back at my desk.

-David

**Pull request merged**